### PR TITLE
Restore "go to channel monitor" Button

### PR DESCRIPTION
**Pull request by eshifri**
Summary of changes:
This PR brings back the "go to channel monitor" button, that was present in 2.7.

I would not insist on it, but I think it has some use.
For the TXs with buttons it is not really necessary, since this functionality is available with MDL button, but for the buttonless radios I think it adds some convenience.

### CODE REVIEW

**Comment by pfeerick** [comment]
Personally think it was merely oversight - as you say it was a convenience for touch screen-only radios, and gave you the overview for all mixes and LS at any page.

**Comment by raphaelcoeffic** [comment]
May I suggest to have this only on radios that really do not have keys? Otherwise it might feel odd to not being able to access that button via encoder wheel.

File: radio/src/gui/colorlcd/menu_model.cpp [hunk 1]
```diff
@@ -74,3 +75,13 @@ void ModelMenu::onEvent(event_t event)
...
+
+void ModelMenu::addButton(TabsGroupHeader* header)
+{
+  OpenTxTheme::instance()->createTextButton(
+      header, {LCD_W / 2 + 6, MENU_TITLE_TOP + 1, LCD_W / 2 - 8, MENU_TITLE_HEIGHT - 2},
+      STR_OPEN_CHANNEL_MONITORS, [=]() {
+        pushEvent(EVT_KEY_FIRST(KEY_MODEL));
+        return 0;
+      });
+}
```
Comment (gagarinlg) [inline radio/src/gui/colorlcd/menu_model.cpp:87]: I think the name of that function is very generic for such a specific thing it does.
Maybe add something to the tab carousel to add a button there, with a lambda to trigger an action?

**Review by gagarinlg — CHANGES_REQUESTED** [review]

**Comment by JimB40** [comment]
Not the best moment to bring it but I'd strongly advocate to have all-around-system quick shorcuts (for radios without keys available via quick-menu icons). If we can manage to make SYS,MDL & TELE configurable (eg long TELE can invoke Monitors) this will solve problem and benfit will be more UI space.

**Comment by pfeerick** [comment]
Indeed. That is off topic, and hopefully something for 2.9... but certainly won't be forgotten regardless ;)

**Comment by eshifri** [comment]
@raphaelcoeffic 
This makes sense definitely.
Do we have a define for touch-only TXs?

**Comment by gagarinlg** [comment]
> Do we have a define for touch-only TXs?

No, but for color LCD radios with Keys:
-DHARDWARE_KEYS

so
#if !defined(HARDWARE_KEYS)

should work

**Comment by eshifri** [comment]
@gagarinlg I do not think so: for NV14 HARDWARE_KEYS is defined as well. It is required to re-purpose trim buttons for ENTER/EXIT. Also without this definition events are not handled by onEvent().
So it looks like we need some other way to handle it.

**Comment by pfeerick** [comment]
So perhaps we need something that allows you to query if the TP is actually active/present in conjunction with `HARDWARE_TOUCH`? Which would probably be helpful for other parts of the code that just make assumptions based on the define.

**Comment by eshifri** [comment]
@raphaelcoeffic Actually, this button is accessible with the roller. (Eventually I got TX in my hands. :-) ).
So I think it can be displayed on all color targets. I think it makes sense: if user is using touch screen he can access channel monitor without going to buttons.

**Comment by pfeerick** [comment]
Oh, interesting... so it gets inserted between the last item on the mix list, and the options at the bottom in as far as how you get to it via scrolling... that works. I would probably have liked it to before the first mix / after the add mix button, but it does work. 🤷

File: radio/src/gui/colorlcd/menu_model.cpp (hunk 1 above)
Comment (eshifri) [reply radio/src/gui/colorlcd/menu_model.cpp]: I have changed the name of the function, but I do not think it makes sense to do it with lambda: This functionality is completely internal to ModelMenu and nobody outside it should know about its existence. ModelMenu is constructed in two different places and the same lambda-expression will have to be written twice.
So, especially taking into account that there is no need to differentiate targets with and without HW buttons, I suggest to keep current implementation.

Comment (gagarinlg) [reply radio/src/gui/colorlcd/menu_model.cpp]: I can live with the changed name

**Comment by pfeerick** [comment]
Merging as it is accessible on non-touch only TXs ;)

**Pull request merged**